### PR TITLE
A couple PEP 654 edits

### PR DESCRIPTION
**Pull request by 1st1**
* Fix a few markup issues & typos; use REsT references.
* Mention Task Groups in Motivation; add an example of the ``trio.MultiError`` API

### CODE REVIEW

File: pep-0654.rst
Comment (iritkatriel) [inline pep-0654.rst:1101]:
```suggestion
``with`` block.  We need to explicitly reraise exceptions we are not handling. Handling more exception types or implementing more complex
```

File: pep-0654.rst
Comment (iritkatriel) [inline pep-0654.rst:1088]:
```suggestion
       return exc   # reraise exc
```

**Review by iritkatriel — APPROVED** [review]
Great! I just added the point about having to explicitly reraise exception you're not handling.

**Comment by 1st1** [comment]
Thanks Irit!

@gvanrossum Guido, does the PR look good to you?

File: pep-0654.rst
Comment (gvanrossum) [inline pep-0654.rst:52]: This needs some more elaboration. Maybe just reference the complexity of Trio's solution?

File: pep-0654.rst
Comment (gvanrossum) [inline pep-0654.rst:40]: Maybe also link to `[9]` here?

File: pep-0654.rst
Comment (gvanrossum) [inline pep-0654.rst:103]:
```suggestion
``MultiError`` [9]_ type. However, such an approache requires calling code to catch
the container exception type, and then to inspect it to determine the types of
errors that had occurred, extract the ones it wants to handle, and reraise the
rest.
```

File: pep-0654.rst
Comment (iritkatriel) [inline pep-0654.rst:52]: Is there an open issue or something else you could link to here?

File: pep-0654.rst
Comment (gvanrossum) [inline pep-0654.rst:1064]:
```suggestion
Programming Without ``except *``
```
(Not sure if RsT likes code blocks in titles though.)

File: pep-0654.rst
Comment (gvanrossum) [inline pep-0654.rst:1068]:
```suggestion
Consider the following simple example of the ``except *`` syntax (pretending
Trio natively supported this proposal):
```

(Also: it's a little unconventional to see the "efficient" version first and the "inefficient" version second. Usually we do before/after, where before is the old way and after is the new, improved way. :-)

File: pep-0654.rst
Comment (gvanrossum) [inline pep-0654.rst:1078]: Please use 4-space indents.

File: pep-0654.rst
Comment (gvanrossum) [inline pep-0654.rst:1102]:
```suggestion
``with`` block.  We need to explicitly reraise exceptions we are not handling.
Handling more exception types or implementing more complex
exception handling logic will only further complicate the code to the point
```

**Review by gvanrossum — COMMENTED** [review]
(Sorry I seem to be a bit hung up on tenses. :-)

File: pep-0654.rst
Comment (1st1) [reply pep-0654.rst:1064]: It should handle them OK, but they will look awkward when rendered to HTML. So I personally try to avoid using them in titles.

**Comment by gvanrossum** [comment]
(Please use a more precise commit message when you're applying a suggestion or using the wed editor. "Update <document>" says nothing.)

File: pep-0654.rst
Comment (gvanrossum) [reply pep-0654.rst:1064]: Maybe use single quotes then?

File: pep-0654.rst
Comment (1st1) [reply pep-0654.rst]: Yeah! I had the same thought, but then opted for my variant because of just how arcane the "inefficient" way looks like compared to `except*`. I'm afraid that readers would be lost in it and skip the whole section, not getting why we added it in the first place. :/

**Comment by 1st1** [comment]
> (Please use a more precise commit message when you're applying a suggestion or using the wed editor. "Update " says nothing.)

Yeah, I'm always folding all suggestion commits manually before merging the PR.

File: pep-0654.rst
Comment (1st1) [reply pep-0654.rst:1064]: Alright

File: pep-0654.rst
Comment (1st1) [reply pep-0654.rst:52]: @iritkatriel @gvanrossum I can add a section to the PEP about TaskGroups API. Or you guys think that creating a bpo with the proposed design would be a better move?

File: pep-0654.rst
Comment (1st1) [reply pep-0654.rst]: Done

**Comment by 1st1** [comment]
I've forced pushed a rebased version with all suggestion commits squashed.

File: pep-0654.rst
Comment (iritkatriel) [reply pep-0654.rst:52]: Adding that design here means we are asking the steering council to approve it as part of this PEP, and if it's not straightforward I think it makes this PEP too big in scope. To Guido's point - isn't this really the same problem/use case as Trio's?

Comment (gvanrossum) [reply pep-0654.rst:52]: Yeah, I think we could shorten this parargaph so it just says "the same applies for asyncio".

**Comment by gvanrossum** [comment]
Also, maybe just land it and iterate in a new PR.

File: pep-0654.rst
Comment (1st1) [reply pep-0654.rst:52]: My problem with removing every notion of TaskGroups from this PEP weakens its motivation.  Ultimately adding task groups to asyncio were the main reason why we started working on this PEP.

Comment (gvanrossum) [reply pep-0654.rst:52]: So say exactly that.

Comment (iritkatriel) [reply pep-0654.rst:52]: I don't think you should remove it, just roll it into the same paragraph with Trio because it's the same kind of use case.  Like the paragraph about multiple use callbacks has two different use cases (atexit and pytest).

Comment (1st1) [reply pep-0654.rst]: I've pushed an update, basically adding this paragraph to the first bullet point:

>   Implementing a better, inspired by Trio nurseries, task spawning API in
  asyncio was the main motivation for this PEP. That work is currently blocked
  on Python not having native language level support for exception groups.

I propose to merge this PR with that, but if you guys want to phrase it differently please open a follow up PR.

**Comment by 1st1** [comment]
> "Implementing a better task spawning API in asyncio, inspired by Trio nurseries, was the main motivation for this PEP."

Done, @iritkatriel

**Comment by 1st1** [comment]
Merged! Thanks for reviews and suggestions @iritkatriel and @gvanrossum. Please feel free to edit the new content.

**Comment by iritkatriel** [comment]
Cool!   I think there are a bunch of section headings that appear at the same level as except* but should be under it. Will fix.

**Comment by 1st1** [comment]
> Cool! I think there are a bunch of section headings that appear at the same level as except* but should be under it. Will fix.

Also need to make sure we use consistent indentation in our code snippets. Right now it's 2 spaces in some -- need to change that to 4.  Do you want me to submit a PR?

**Comment by iritkatriel** [comment]
> Also need to make sure we use consistent indentation in our code snippets. Right now it's 2 spaces in some -- need to change that to 4. Do you want me to submit a PR?

Yes please

**Pull request merged**
